kubernetes/client.go

```diff
@@ -99,30 +101,24 @@ func NewClient() (*IstioClient, error) {
 
 // It returns a list of all namespaces/projects of the cluster.
 // It returns an error on any problem.
-func (in *IstioClient) GetNamespaces() ([]string, error) {
+func (in *IstioClient) GetNamespaces() ([]models.Namespace, error) {
 	namespaces, err := in.k8s.CoreV1().Namespaces().List(emptyListOptions)
 	if err != nil {
 		return nil, err
 	}
-	names := make([]string, len(namespaces.Items))
-	for i, namespace := range namespaces.Items {
-		names[i] = namespace.Name
-	}
-	return names, nil
+
+	return models.CastNamespaceCollection(namespaces), nil
```

Comment:
I guess that to maintain the abstraction layers, these kubernetes/client.go should work within the kubernetes API model and not return our own model used for the REST.
This means that the Cast* call can be performed within the handler, or just cast the output placing these functions in the model.
On this way, I guess we don't mix both models which are related but independent.

On that sense, if things change we maintain the separation between layers.

What do you think ?

Reply:
@lucasponce I was thinking another way around. But I like what you suggested to move it outside of this. I was thinking more of preventing the other layers to receive and spread models that are third-parties.
I do like the approach @aljesusg did on his proposal and leave to the models the responsibility of interacting to the backend.